test/managers/integration/OnOfframpManager.t.sol

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.28;
+
+import "test/spoke/BaseTest.sol";
+import {D18, d18} from "src/misc/types/D18.sol";
+import {IAuth} from "src/misc/interfaces/IAuth.sol";
+import {CastLib} from "src/misc/libraries/CastLib.sol";
+import {IERC7751} from "src/misc/interfaces/IERC7751.sol";
+
+import {AssetId} from "src/common/types/AssetId.sol";
+import {ShareClassId} from "src/common/types/ShareClassId.sol";
+
+import {UpdateContractMessageLib} from "src/spoke/libraries/UpdateContractMessageLib.sol";
+
+import {UpdateRestrictionMessageLib} from "src/hooks/libraries/UpdateRestrictionMessageLib.sol";
+
+import {OnOfframpManager} from "src/managers/OnOfframpManager.sol";
+import {IOnOfframpManager} from "src/managers/interfaces/IOnOfframpManager.sol";
+
+abstract contract OnOfframpManagerBaseTest is BaseTest {
```

Comment:
I think it can be easier to test and understand the test without all the IT overhead (and also do future refactors) in a UT environment, just by mocking deposit/withdraw from `BalanceSheet`

Reply:
Yeah I need to check that, but would still argue on this level integration tests are critical, and potentially more important than mocked unit tests. E.g. I wouldn't have caught the missing approval without it.